app/controllers/catalog_controller.rb

```diff
@@ -202,6 +202,38 @@ def expire_session
       a.params.dig(:f, :format_f)&.include?('Database & Article Index')
     }
 
+    # Some filters (e.g., subject_f) are capable of driving meaning correlations;
+    # others are not, and either generate spurious correlations, or at best pointlessly add extra
+    # overhead to Solr request.
+    # Keys below indicate filters that we should not attempt to use for purpose of cacluating
+    # correlations -- either entirely (nil value) or for an array of certain filter values
+    @@CORRELATION_BLACKLIST = {
+      :access_f => nil,
+      :record_source_f => nil,
+      :format_f => ['Database & Article Index']
+    }
+
+    actionable_filters = lambda { |a, b, c|
+      params = a.params
+      return true if params[:q].present?
+      f = params[:f]
+      return false if f.nil?
+      # we return true if there is at least one non-blacklisted filter
+      return true if f.size > @@CORRELATION_BLACKLIST.size
+      f.keys.map(&:to_sym).each do |key|
```

Comment:
i suggest `f.symbolize_keys.each do |facet_field, facet_values|` for improved readability

Reply:
actually, f is an object that inherits from `HashWithIndifferentAccess`, so the `symbolize_keys` is not needed here:

Reply:
That's what I expected initially, but it wasn't working until I added that in! Maybe I was doing something wrong, or maybe the exact implementation is version-dependent? On my local machine I'm using 2.3.1 ...

Reply:
yeah, i see, because Hash _access_ is indifferent but this iterator will yield the original key (a String) to the block, and that will impact the access to values in @@CORRELATION...

 if you use `.each` on the Hash itself you can avoid the ambiguity:

```
      f.symbolize_keys.each do |facet_field, facet_values|
        return true unless @@CORRELATION_IGNORELIST.include?(facet_field)
        ignorelisted_vals = @@CORRELATION_IGNORELIST[facet_field]
        next unless ignorelisted_vals
        # then there exist some *non*-ignorelisted vals that would make filters actionable
        # check for such vals here
        return true unless (facet_values & ignorelisted_vals).any?
      end
```

Reply:
ah yeah, so it's whether `@@correlation` is "indifferent" that's relevant, and it's not indifferent. Anyway just pushed a commit incorporating these comments ... much improved, thanks!